packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsSteps.tsx

```diff
@@ -27,6 +27,12 @@ import { DepositCountdown } from '../common/DepositCountdown'
 import { useRemainingTime } from '../../state/cctpState'
 import { isDepositReadyToRedeem } from '../../state/app/utils'
 import { Address } from '../../util/AddressUtils'
+import { isTeleport } from '../../token-bridge-sdk/teleport'
+import {
+  firstRetryableLegRequiresRedeem,
```

Comment:
> But if we transfer from L2 to L3, l2L3RetryableRequiresRedeem will never be true because it checks l2ToL3MsgData which will always be undefined here.

Actually this is still a naming problem. For L2 to L3 transfers also we check `l1ToL2MsgData`. This will be solved when we properly rename the objects to signal parent-to-child-msgData.

Also as far as the usage goes, `firstRetryableLegRequiresRedeem(tx)` will return true if the deposit (standard or teleport) requires redemption. So it holds true for both the cases. For the usage of `l2L3RetryableRequiresRedeem` I have updated it's name to `secondRetryableLegForTeleportRequiresRedeem` so that it's understandable, and in code, it's always being called for isTeleport txns and nowhere else.

Reply:
That's nice